src/component/Pagination.jsx

```diff
@@ -1,24 +1,95 @@
 // Pagination.js
-const Pagination = ({ pagebuttons, activePage, handlePageClick }) => {
+import styled from "styled-components";
+import { useState } from "react";
+
+const PaginationContainer = styled.div`
+  display: flex;
+  margin: 0 auto;
+  width: 304px;
+  height: 40px;
+  align-items: center;
+`;
+
+const PaginationButton = styled.button`
+  border-radius: 50px;
+  width: 40px;
+  height: 40px;
+  background-color: #ffffff;
+  border: 1px solid #e5e7e8;
+  margin-left: 4px;
+  margin-top: 40px;
+  font-size: 16px;
+  line-height: 26px;
+  font-weight: 600;
+
+  &:hover {
+    background-color: #f0f0f0;
+    border-color: #ccc;
+  }
+`;
+
+const Pagination = ({ activePage, setActivePage, totalpage }) => {
+  const [pagebuttons, setPagebuttons] = useState([1, 2, 3, 4, 5]);
+
+  const handleNextClick = () => {
+    console.log("totalpage:", totalpage);
+    if (activePage >= totalpage) {
+      return;
+    }
+    if (activePage % 5 === 0) {
+      pagebuttons.forEach((num, index) => {
+        pagebuttons[index] = activePage + index + 1;
+      });
+      setActivePage(activePage + 1);
+      setPagebuttons(pagebuttons);
+    } else {
+      setActivePage(activePage + 1);
+    }
+  };
+
+  const handlePreviousClick = () => {
+    if (activePage === 1) {
+      return;
+    }
+    if ((activePage - 1) % 5 === 0) {
+      pagebuttons.forEach((num, index) => {
+        pagebuttons[index] = activePage - index - 1;
+      });
+      pagebuttons.reverse();
+      setActivePage(activePage - 1);
+      setPagebuttons(pagebuttons);
+    } else {
+      setActivePage(activePage - 1);
+    }
+  };
+
+  const handlePageClick = (number) => {
+    setActivePage(number);
+  };
+
   const getButtonStyle = (number) => ({
-    backgroundColor: activePage === number ? '#2F80ED' : 'white',
-    color: activePage === number ? 'white' : 'black',
+    backgroundColor: activePage === number ? "#2F80ED" : "white",
+    color: activePage === number ? "white" : "black",
   });
 
   return (
-    <div className="pagination">
-      <button>&lt;</button>
+    <PaginationContainer>
+      <PaginationButton onClick={() => handlePreviousClick()} type="button">
```

Comment:
P1:
해당 함수에 arg가 필요없다면 아래처럼 사용해주세요.

```suggestion
      <PaginationButton onClick={handlePreviousClick} type="button">
```